sound/soc/sof/ipc4-topology.c

```diff
@@ -1061,15 +1061,117 @@ static bool sof_ipc4_is_single_format(struct snd_sof_dev *sdev,
 	return true;
 }
 
+static bool sof_ipc4_copier_is_single_format(struct snd_sof_dev *sdev,
+					     struct sof_ipc4_pin_format *pin_fmts,
+					     u32 pin_fmts_size)
+{
+	struct sof_ipc4_audio_format *fmt;
+	u32 channels, valid_bits;
+	int i;
+
+	fmt = &pin_fmts[0].audio_fmt;
+	channels = SOF_IPC4_AUDIO_FORMAT_CFG_CHANNELS_COUNT(fmt->fmt_cfg);
+	valid_bits = SOF_IPC4_AUDIO_FORMAT_CFG_V_BIT_DEPTH(fmt->fmt_cfg);
+
+	/* check if all output formats in topology are the same */
+	for (i = 1; i < pin_fmts_size; i++) {
```

Comment:
Not clear how this applies just to output formats only? if this is a host copier on capture what would this mean?